core/src/main/java/com/linecorp/armeria/server/websocket/WebSocketServiceBuilder.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.websocket;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.Flags;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.common.websocket.WebSocketDecoderConfig;
+import com.linecorp.armeria.common.websocket.WebSocketFrame;
+
+/**
+ * Builds a {@link WebSocketService}.
+ */
+@UnstableApi
+public final class WebSocketServiceBuilder {
+
+    private static final WebSocketDecoderConfig defaultDecoderConfig = WebSocketDecoderConfig.builder().build();
+
+    private final WebSocketHandler handler;
+    private WebSocketDecoderConfig decoderConfig = defaultDecoderConfig;
+    private Set<String> subprotocols = ImmutableSet.of();
+    private Set<String> allowedOrigins = ImmutableSet.of();
+    private long closeTimeoutMillis = Flags.defaultRequestTimeoutMillis();
+
+    WebSocketServiceBuilder(WebSocketHandler handler) {
+        this.handler = requireNonNull(handler, "handler");
+    }
+
+    /**
+     * Sets the {@link WebSocketDecoderConfig} to decode {@link WebSocketFrame}s.
+     */
+    public WebSocketServiceBuilder decoderConfig(WebSocketDecoderConfig decoderConfig) {
```

Comment:
How about adding all decoder settings method to this builder directly, so a user doesn't have to build two objects?